lib/history/sqlite/sqlite.go

```diff
@@ -30,115 +29,142 @@ import (
-		database: database,
-		// TODO: store and recover lastStatus in case satellite agent restarts.
-		lastStatus: nil,
-	}, nil
+	// if err := timeline.initPrevStatus(ctx); err != nil {
+	// 	return nil, trace.Wrap(err, "failed to init previous status")
+	// }
+
+	// TODO: should this be called in constructor method?
+	// Let caller run event eviction loop?
+	// Create separate init function?
+	go timeline.eventEvictionLoop()
+
+	return timeline, nil
 }
 
-// initSQLite initializes connection to database provided by dbPath and
-// initializes `events` table.
-func initSQLite(ctx context.Context, dbPath string) (*sqlx.DB, error) {
-	database, err := sqlx.ConnectContext(ctx, "sqlite3", dbPath)
+// initSQLite initializes connection to database and initializes `events` table.
+func (t *Timeline) initSQLite(ctx context.Context) error {
+	database, err := sqlx.ConnectContext(ctx, "sqlite3", t.config.DBPath)
 	if err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
 
 	if _, err := database.ExecContext(ctx, createTableEvents); err != nil {
-		return nil, trace.Wrap(err)
+		return trace.Wrap(err)
 	}
 
-	return database, nil
+	t.database = database
+	return nil
+}
+
+// initPrevStatus initializes the previously record status in case satellite
+// is restarted.
+func (t *Timeline) initPrevStatus(ctx context.Context) error {
+	// TODO
+	return trace.NotImplemented("not implemented")
 }
 
-// getSize returns the current number of rows stored in the provided database.
-func getSize(ctx context.Context, database *sqlx.DB) (size int, err error) {
-	row := database.QueryRowContext(ctx, "SELECT COUNT(*) FROM events")
-	if err := row.Scan(&size); err != nil {
-		return -1, trace.Wrap(err)
+// eventEvictionLoop periodically evicts old events to free up storage.
+func (t *Timeline) eventEvictionLoop() {
+	ticker := t.config.Clock.NewTicker(evictionFrequency)
+	for {
+		<-ticker.Chan()
+
+		// All events before this time will be deleted.
+		retentionCutOff := t.config.Clock.Now().Add(-(t.config.RetentionDuration))
+
+		ctx, cancel := context.WithTimeout(context.Background(), evictionTimeout)
+		if err := t.evictEvents(ctx, retentionCutOff); err != nil {
+			cancel()
+			log.WithError(err).Warnf("Error evicting expired events.")
+		}
+		cancel()
 	}
-	return size, nil
 }
 
 // RecordStatus records the differences between the previously stored status and
 // the provided status.
-func (t *Timeline) RecordStatus(ctx context.Context, status *pb.SystemStatus) (err error) {
-	t.mu.Lock()
+func (t *Timeline) RecordStatus(ctx context.Context, status *pb.NodeStatus) (err error) {
+	t.statusMutex.Lock()
 
-	events := history.DiffCluster(t.Config.Clock, t.lastStatus, status)
+	events := history.DiffNode(t.config.Clock, t.lastStatus, status)
 	if len(events) == 0 {
-		t.mu.Unlock()
+		t.statusMutex.Unlock()
 		return nil
 	}
-
-	t.size = t.size + len(events)
-	if t.size > t.Config.Capacity {
-		t.size = t.Config.Capacity
-	}
 	t.lastStatus = status
```

Comment:
Nit: I usually try to add a bit more debugging information to ease troubleshooting later on or just be able to see what's going on, so for example if a status changes, I would log the events or something.